docs/storage/volume_migration.md

```diff
@@ -0,0 +1,204 @@
+The destination volume should be equal to or larger than the source volume. However, the additional difference in the size of the destination volume is not instantly visible within the VM and must be manually resized because the guest is unaware of the migration.
+
+
+## Datavolume template
+
+Updating a datavolume that is referenced by a datavolume template requires special caution. The volumes section must include a reference to the name of the datavolume template. This means that the datavolume templates must either be entirely deleted or updated as well.
+
+Example of updating the datavolume for the original VM in the first example:
+```yaml
+apiVersion: cdi.kubevirt.io/v1beta1
+kind: DataVolume
+metadata:
+  name: dst-dv
+spec:
+  source:
+      blank: {}
+  pvc:
+    accessModes:
+      - ReadWriteOnce
+    resources:
+      requests:
+        storage: 2Gi
+    storageClassName: local
+---
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    kubevirt.io/vm: vm-dv
+  name: vm-dv
+spec:
+  updateVolumesStrategy: Migration
+  dataVolumeTemplates:
+  - metadata:
+      name: dst-dv
+    spec:
+      pvc:
+        accessModes:
+        - ReadWriteOnce
+        volumeMode: Filesystem
+        resources:
+          requests:
+            storage: 2Gi
+        storageClassName: local
+      source:
+        registry:
+          url: docker://registry:5000/kubevirt/alpine-container-disk-demo:devel
+  running: true
+  template:
+    metadata:
+      labels:
+        kubevirt.io/vm: vm-dv
+    spec:
+      domain:
+        devices:
+          disks:
+          - disk:
+              bus: virtio
+            name: datavolumedisk1
+          interfaces:
+          - masquerade: {}
+            name: default
+        resources:
+          requests:
+            memory: 128Mi
+      terminationGracePeriodSeconds: 0
+      volumes:
+      - dataVolume:
+          name: dst-dv
+        name: datavolumedisk1
+      networks:
+      - name: default
+        pod: {}
+```
+
+## Limitations
+
+Only certain types of disks and volumes are supported to be migrated. For an invalid type of volume the RestartRequired condition is set and volumes will be replaced upon VM restart.
+Currently, the volume migration is supported between PersistentVolumeClaims and Datavolumes.
+Additionally, volume migration is forbidden if the volumes is:
+* shareable, since it cannot guarantee the data consistency with multiple writers
+* hotpluggable, this case isn't currently supported
+* filesystem, since virtiofs doesn't curretly support live-migration
+* lun, originally the disk might support SCSI protocol but the destination PVC class does not. This case isn't currently supported.
+
+Currently, KubeVirt only enables live migration between separate nodes. Volume migration relies on live migration; hence, warm migrating storage on the same node is also not possible.
```

Comment:
So there are 3 types of migration:
1. Cold. The VM is turned off, and the disks are migrated, then the VM is turned on.
2. Warm. The VM is running, and a snapshot of the disk is copied to the target. Then the VM is turned off and the remaining delta between the source and target is copied, then the VM is turned on again. This should be much faster than a cold migration as there is less data to copy while the VM is turned off.
3. Hot (or live) migration. This what this document is about. So I would change the wording according to the suggestion.
```suggestion
Currently, KubeVirt only enables live migration between separate nodes. Volume migration relies on live migration; hence, live migrating storage on the same node is also not possible.
```

Reply:
Replaced, thanks!